### PR TITLE
New script - Get list row

### PR DESCRIPTION
**Pull request by ChanochShayner**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/26081

## Description
Parse list by header and value or parse all the list

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Demisto
any

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [x] Tests
- [ ] Documentation

### CODE REVIEW

**Comment by ronykoz** [comment]
Didn't go over the code, but please move this to the CommonScripts pack

File: Packs/GetListRow/Scripts/GetListRow/GetListRow.py [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+        return_error("Error: The supplied list name not found.")
+
+
+def validate_header_exists(headers, header):
+    if header not in headers:
+        return_error("Error: The supplied header name not found.")
+
+
+def list_to_headers_and_lines(list_result):
+    lines_without_headers = [line.split(',') for line in list_result.split('\n')]
```
Comment (guykeller) [inline Packs/GetListRow/Scripts/GetListRow/GetListRow.py:21]: This is with the  headers.
lines_without_headers should start from index 1, like so: `[1:]`

File: Packs/GetListRow/Scripts/GetListRow/GetListRow.py [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    headers = lines_without_headers[0]
+    return lines_without_headers[1:], headers
+
+
+def list_to_context_format(list_lines, headers):
```
Comment (guykeller) [inline Packs/GetListRow/Scripts/GetListRow/GetListRow.py:26]:
```suggestion
def lines_to_context_format(list_lines, headers):
```

File: Packs/GetListRow/Scripts/GetListRow/GetListRow.py [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    result = []
```
Comment (guykeller) [inline Packs/GetListRow/Scripts/GetListRow/GetListRow.py:27]: let's rename to something more clear, like `lines_context`

File: Packs/GetListRow/Scripts/GetListRow/GetListRow.py [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+    return result
+
+
+def parse_relevant_rows(headers, lines, header, value, context, parse_all=False):
+    if parse_all:
+        result = list_to_context_format(lines, headers)
+    else:
+        header_location = headers.index(header)
+        specific_lines_to_parse = [line for line in lines if line[header_location] == value]
+        result = list_to_context_format(specific_lines_to_parse, headers)
```
Comment (guykeller) [inline Packs/GetListRow/Scripts/GetListRow/GetListRow.py:40]: rename `result` please, it isn't an informative name

File: Packs/GetListRow/Scripts/GetListRow/GetListRow.py [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+    if not result:
+        return CommandResults(
+            readable_output="No results found"
+        )
+    context["results"] = result
+    markdown = tableToMarkdown('List Result', result, headers=headers)
```
Comment (guykeller) [inline Packs/GetListRow/Scripts/GetListRow/GetListRow.py:46]:
```suggestion
    human_readable = tableToMarkdown('List Result', result, headers=headers)
```

File: Packs/GetListRow/Scripts/GetListRow/GetListRow.py [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+        outputs_prefix='''getListRow.Result(val.header && val.header == obj.header
+                          && val.value == obj.value && val.list_name == obj.list_name)''',
+        outputs=context,
+        readable_output=markdown
+    )
+
+
+def parse_list(parse_all, header, value, list_name):
+    validate_args(parse_all, header, value)
+    list_result = demisto.executeCommand("getList", {'listName': list_name})[0]['Contents']
```
Comment (guykeller) [inline Packs/GetListRow/Scripts/GetListRow/GetListRow.py:57]: let's rename to `list`

File: Packs/GetListRow/Scripts/GetListRow/GetListRow.py [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+    context = {
+        "list_name": list_name,
+        "header": header,
+        "value": value,
+    }
+    validate_list_exists(list_result)
+    lines, headers = list_to_headers_and_lines(list_result)
```
Comment (guykeller) [inline Packs/GetListRow/Scripts/GetListRow/GetListRow.py:64]: let's change the order of lines and  headers
if the script is called `list_to_headers_and_lines` it should return headers, lines

File: Packs/CommonScripts/Scripts/GetlistRow/GetListRow.py [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+        header_location = headers.index(header)
+        specific_lines_to_parse = [line for line in lines if line[header_location] == value]
+        lines_context = lines_to_context_format(specific_lines_to_parse, headers)
+        output_condition = '''getListRow.Result(val.header && val.list_name == obj.list_name &&
+                           val.header == obj.header && val.value == obj.value && val.parse_all == obj.parse_all)'''
+    if not lines_context:
+        return CommandResults(
+            readable_output="No results found"
+        )
+    context["lines"] = lines_context
```
Comment (ronykoz) [inline Packs/CommonScripts/Scripts/GetlistRow/GetListRow.py:48]: i think it should be 
```suggestion
    context["Results"] = lines_context
```

File: Packs/CommonScripts/Scripts/GetlistRow/GetListRow.py [hunk 9]
```diff
@@ -0,0 +1,87 @@
...
+
+def parse_list(parse_all, header, value, list_name):
+    validate_args(parse_all, header, value)
+    list_data = demisto.executeCommand("getList", {'listName': list_name})[0]['Contents']
+    context = {
+        "list_name": list_name,
+        "header": header,
+        "value": value,
+        "parse_all": parse_all
+    }
```
Comment (ronykoz) [inline Packs/CommonScripts/Scripts/GetlistRow/GetListRow.py:65]: this list should match the output from the `.yml` file, please update the strings here to be capitalized
and update the outputs with the parse_all

File: Packs/CommonScripts/Scripts/GetlistRow/GetListRow.yml [hunk 10]
```diff
@@ -0,0 +1,60 @@
...
+  required: false
+  secret: false
+comment: Parse list by header and value
+commonfields:
+  id: getListRow
+  version: -1
+enabled: true
+name: getListRow
+outputs:
+- contextPath: GetListRow.Header
```
Comment (ronykoz) [inline Packs/CommonScripts/Scripts/GetlistRow/GetListRow.yml:38]:
```suggestion
- contextPath: GetListRow.Result.Header
```

File: Packs/CommonScripts/Scripts/GetlistRow/GetListRow.py (hunk 9 above)
Comment (ronykoz) [reply Packs/CommonScripts/Scripts/GetlistRow/GetListRow.py:65]: List_Name
Header
Value
Parse_All

File: Packs/CommonScripts/Scripts/GetlistRow/GetListRow.py [hunk 11]
```diff
@@ -0,0 +1,87 @@
...
+
+def parse_relevant_rows(headers, lines, header, value, context, parse_all=False):
+    if parse_all:
+        lines_context = lines_to_context_format(lines, headers)
+        output_condition = "getListRow.Result(val.list_name == obj.list_name && val.parse_all == obj.parse_all)"
+    else:
+        header_location = headers.index(header)
+        specific_lines_to_parse = [line for line in lines if line[header_location] == value]
+        lines_context = lines_to_context_format(specific_lines_to_parse, headers)
+        output_condition = '''getListRow.Result(val.header && val.list_name == obj.list_name &&
```
Comment (ronykoz) [inline Packs/CommonScripts/Scripts/GetlistRow/GetListRow.py:42]:
```suggestion
        output_condition = '''GetListRow.Result(val.header && val.list_name == obj.list_name &&
```

File: Packs/CommonScripts/Scripts/GetlistRow/GetListRow.py [hunk 12]
```diff
@@ -0,0 +1,87 @@
...
+        header_location = headers.index(header)
+        specific_lines_to_parse = [line for line in lines if line[header_location] == value]
+        lines_context = lines_to_context_format(specific_lines_to_parse, headers)
+        output_condition = '''GetListRow.Result(val.header && val.list_name == obj.list_name &&
+                           val.header == obj.header && val.value == obj.value && val.parse_all == obj.parse_all)'''
+    if not lines_context:
+        return CommandResults(
+            readable_output="No results found"
+        )
+    context["Result"] = lines_context
```
Comment (ronykoz) [inline Packs/CommonScripts/Scripts/GetlistRow/GetListRow.py:48]:
```suggestion
    context["Results"] = lines_context
```

File: Packs/CommonScripts/Scripts/GetlistRow/GetListRow.yml [hunk 13]
```diff
@@ -0,0 +1,60 @@
...
+- contextPath: GetListRow.Result.Header
+  description: Header to search in.
+  type: String
+- contextPath: GetListRow.Result.Value
+  description: Value to search for.
+  type: String
+- contextPath: GetListRow.Result.List_Name
+  description: The list name to search in.
+  type: String
+- contextPath: GetListRow.Results.Parse_All
```
Comment (ronykoz) [inline Packs/CommonScripts/Scripts/GetlistRow/GetListRow.yml:47]:
```suggestion
- contextPath: GetListRow.Result.Parse_All
```

File: Packs/CommonScripts/Scripts/GetlistRow/GetListRow.yml [hunk 14]
```diff
@@ -0,0 +1,60 @@
...
+comment: Parse list by header and value.
+commonfields:
+  id: GetListRow
+  version: -1
+enabled: true
+name: GetListRow
+outputs:
+- contextPath: GetListRow.Result.Header
+  description: Header to search in.
+  type: String
```
Comment (ronykoz) [inline Packs/CommonScripts/Scripts/GetlistRow/GetListRow.yml:40]: you should add another one with 
```
GetListRow.Reults
```

File: Packs/CommonScripts/Scripts/GetlistRow/GetListRow.py [hunk 15]
```diff
@@ -0,0 +1,87 @@
...
+
+def parse_relevant_rows(headers, lines, header, value, context, parse_all=False):
+    if parse_all:
+        lines_context = lines_to_context_format(lines, headers)
+        output_condition = "getListRow.Result(val.list_name == obj.list_name && val.parse_all == obj.parse_all)"
+    else:
+        header_location = headers.index(header)
+        specific_lines_to_parse = [line for line in lines if line[header_location] == value]
+        lines_context = lines_to_context_format(specific_lines_to_parse, headers)
+        output_condition = '''GetListRow.Result(val.header && val.list_name == obj.list_name &&
```
Comment (ronykoz) [inline Packs/CommonScripts/Scripts/GetlistRow/GetListRow.py:42]:
```suggestion
        output_condition = '''GetListRow(val.header && val.list_name == obj.list_name &&
```

Sorry for the late comment about this, but the `Result` is not really needed here, update the outputs as well

File: Packs/CommonScripts/Scripts/GetlistRow/GetListRow.yml [hunk 16]
```diff
@@ -0,0 +1,63 @@
...
+  description: Value to search for.
+  type: String
+- contextPath: GetListRow.Result.ListName
+  description: The name of the list to search in.
+  type: String
+- contextPath: GetListRow.Result.ParseAll
+  description: Parse all the list into the context.
+  type: String
+- contextPath: GetListRow.Result
+  description: All result.
```
Comment (guykeller) [inline Packs/CommonScripts/Scripts/GetlistRow/GetListRow.yml:51]: Let's improve this description

File: Packs/CommonScripts/Scripts/GetlistRow/GetListRow.yml [hunk 17]
```diff
@@ -0,0 +1,63 @@
+args:
+- default: false
+  description: Value to search for (You also need to enter header).
+  isArray: false
+  name: value
+  required: false
+  secret: false
+- default: false
+  description: Header to search in (You also need to enter value)/
```
Comment (guykeller) [inline Packs/CommonScripts/Scripts/GetlistRow/GetListRow.yml:9]:
```suggestion
  description: Header to filter lines by. If the header is specified, a requested value for that header must be specified as well.
```

File: Packs/CommonScripts/Scripts/GetlistRow/GetListRow.yml [hunk 18]
```diff
@@ -0,0 +1,63 @@
+args:
+- default: false
+  description: Value to search for (You also need to enter header).
```
Comment (guykeller) [inline Packs/CommonScripts/Scripts/GetlistRow/GetListRow.yml:3]:
```suggestion
  description: Value to search for. If the value is specified, a specific header must be supplied as well.
```

File: Packs/CommonScripts/Scripts/GetlistRow/GetListRow.yml [hunk 19]
```diff
@@ -0,0 +1,64 @@
...
+  description: Header to search in.
+  type: String
+- contextPath: GetListRow.Result.Value
+  description: Value to search for.
+  type: String
+- contextPath: GetListRow.Result.ListName
+  description: The name of the list to search in.
+  type: String
+- contextPath: GetListRow.Result.ParseAll
+  description: Parse all the list into the context.
```
Comment (ronykoz) [inline Packs/CommonScripts/Scripts/GetlistRow/GetListRow.yml:49]: remove the `Result` from the path, it's not needed

File: Packs/CommonScripts/Scripts/GetlistRow/GetListRow.yml [hunk 20]
```diff
@@ -0,0 +1,64 @@
...
+  type: String
+- contextPath: GetListRow.Result
```
Comment (ronykoz) [inline Packs/CommonScripts/Scripts/GetlistRow/GetListRow.yml:51]:
```suggestion
- contextPath: GetListRow.Results
```

**Comment by kirbles19** [comment]
@ronykoz Done.

**Pull request merged**